pages/collaborators.tsx

```diff
@@ -0,0 +1,230 @@
+import { useState, useMemo, useRef, useEffect } from "react"
+import { NextSeo } from "next-seo"
+import { UserLists } from "../components/UserLists"
+import { useUserId } from "../lib/web/utils"
+import { Tournaments } from "../components/Tournaments"
+import { api } from "../lib/web/trpc"
+import { TeamListContent } from "../components/TeamListContent"
+import { TournamentContent } from "../components/TournamentContent"
+import { signInToFatebook } from "../lib/web/utils"
+import {
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  ChevronDoubleDownIcon,
+  ChevronDoubleUpIcon,
+} from "@heroicons/react/20/solid"
+import {
+  ChevronLeftIcon as ChevronLeftIconLarge,
+  ChevronRightIcon as ChevronRightIconLarge,
+} from "@heroicons/react/24/solid"
+
+export default function CollaboratorsPage() {
+  const userId = useUserId()
+  const [currentIndex, setCurrentIndex] = useState(0)
+  const containerRef = useRef<HTMLDivElement>(null)
+  const [isOverlayVisible, setIsOverlayVisible] = useState(false)
+  const [overlayTop, setOverlayTop] = useState("4rem")
+
+  const tournamentsQ = api.tournament.getAll.useQuery({})
+  const userListsQ = api.userList.getUserLists.useQuery()
+
+  type CarouselItemType =
+    | { type: "userList"; id: string; name: string; data: any }
+    | { type: "tournament"; id: string; name: string; data: any }
```

Comment:
Better to have data be the actual expected type too btw, this is another reason to let TS infer the type from the TRPC procedure rather than defining our own type